### PR TITLE
Chore: Refactor usage of legacy data contracts

### PR DESCRIPTION
**Pull request by marefr**
**What this PR does / why we need it**:
Refactor usage of legacy data contracts. Moves legacy contracts from the /pkg/plugins package to legacydata package (/pkg/tsdb/legacydata?). Goal is to make the legacy part contained in one package and make it easy to understand what is legacy and what is not. Should make it easier to migrate away from using the legacy contracts and clear what part are still  dependent on it.

Structure after refactor:
image

- [x] Move legacy data contracts to legacydata package
- [x] Refactor pkg/expr to be a proper service/dependency that can be provided to wire to remove some weird dependencies to SSE in ngalert and other places.
- [x] Refactor pkg/expr to not use the legacydata,RequestHandler and use backend.QueryDataHandler instead.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/tsdb/cloudwatch/live.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func (r *logQueryRunner) publishResults(orgID int64, channelName string) error {
 
 // executeLiveLogQuery executes a CloudWatch Logs query with live updates over WebSocket.
 // A WebSocket channel is created, which goroutines send responses over.
-//nolint: staticcheck // plugins.DataResponse deprecated
+//nolint: staticcheck // legacydata.DataResponse deprecated
```
Comment (wbrowne) [inline pkg/tsdb/cloudwatch/live.go:110]: `//nolint: staticcheck // legacydata.DataResponse deprecated` aren't relevant in this file anymore, right?

File: pkg/tsdb/data_plugin_adapter.go [hunk 2]
```diff
@@ -1,110 +0,0 @@
...
-	"github.com/grafana/grafana-plugin-sdk-go/backend"
-	"github.com/grafana/grafana/pkg/models"
-	"github.com/grafana/grafana/pkg/plugins"
-	"github.com/grafana/grafana/pkg/plugins/adapters"
-	"github.com/grafana/grafana/pkg/services/datasources"
-	"github.com/grafana/grafana/pkg/services/oauthtoken"
-)
-
-// nolint:staticcheck // plugins.DataQuery deprecated
-func dataPluginQueryAdapter(pluginID string, handler backend.QueryDataHandler, oAuthService oauthtoken.OAuthTokenService,
```
Comment (wbrowne) [inline pkg/tsdb/data_plugin_adapter.go:17]: rip

File: pkg/tsdb/sqleng/sql_engine.go [hunk 3]
```diff
@@ -382,7 +382,7 @@ var Interpolate = func(query backend.DataQuery, timeRange backend.TimeRange, tim
 	return sql, nil
 }
 
-//nolint: staticcheck // plugins.DataPlugin deprecated
+//nolint: staticcheck // legacydata.DataPlugin deprecated
```
Comment (wbrowne) [inline pkg/tsdb/sqleng/sql_engine.go:385]: These also seem unnecessary now?

**Review by wbrowne — COMMENTED** [review]
Nice one! 🎉  Spotted a few things worth mentioning, but nothing blocking

**Review by sunker — APPROVED** [review]
Nice! 

I've created an [issue](https://github.com/grafana/grafana/issues/41452) for removing dependencies to legacydata in the cloud data sources.

**Review by wbrowne — APPROVED** [review]
LGTM 🎉

**Review by gotjosh — APPROVED** [review]
LGTM, but would love if @papagian can take a quick peek.

**Review by papagian — APPROVED** [review]
LGTM and I have tested alerting (both legacy and new) and work fine.

**Pull request merged**